pkg/build/webhook/bitbucket/bitbucket_test.go

```diff
@@ -327,13 +376,92 @@ func TestExtractProvidesValidBuildForAPushEventOtherThanMaster(t *testing.T) {
 	}
 }
 
+func TestExtractProvidesValidBuildForAPushEventOtherThanMaster54(t *testing.T) {
+	//setup
+	context := setup(t, "pushevent54-not-master.json", "repo:refs_changed", "other")
+	//execute
+	revision, _, _, proceed, err := context.plugin.Extract(context.buildCfg, "secret100", context.path, context.req)
+
+	//validation
+	if err != nil {
+		t.Errorf("Error while extracting build info: %s", err)
+	}
+	if !proceed {
+		t.Errorf("The 'proceed' return value should equal 'true' %t", proceed)
+	}
+	if revision == nil {
+		t.Fatal("Expecting the revision to not be nil")
+	}
+	if revision.Git.Commit != "178864a7d521b6f5e720b386b2c2b0ef8563e0dc" {
+		t.Error("Expecting the revision to contain the commit id from the push event")
+	}
+}
+
 func TestExtractSkipsBuildForUnmatchedBranches(t *testing.T) {
 	//setup
-	context := setup(t, "pushevent.json", "push", "wrongref")
+	context := setup(t, "pushevent.json", "repo:push", "wrongref")
```

Comment:
@gabemontero i think this was a mistake in the test?  it wasn't checking for an error so the test was passing, but proceed was false because the event type was wrong, not because the branch didn't match.

Reply:
Unfortunately, Ryan did the bitbucket stuff (whereas I did gitlab).

That said, the analogous gitlab test looks exactly the same ... I'm pretty sure the event type is right in that case, but perhaps it needs the same err check as well

Bottom line, something missed during code review unfortunately.